src/status_icons.c

```diff
@@ -0,0 +1,1272 @@
+                    elementId = icon->boostPartner.activeElementID;
+                    set_hud_element_render_pos(elementId, screenX + 2, screenY - 12);
+                    draw_hud_element_2(elementId);
+                }
+            } else
+                goto BOOSTPARTNER;
+        }
+BOOSTPARTNERBACK:
+        if (icon->surprise.active) {
+            if (!(icon->flags & STATUS_ICON_FLAG_SURPRISE)) {
+                if (icon->flags & STATUS_ICON_FLAG_BATTLE || gGameStatusPtr->isBattle != 1) {
+                    clear_hud_element_flags(icon->surprise.activeElementID, 2);
+
+                    x = icon->worldPos.x;
+                    z = icon->worldPos.z;
+
+                    get_screen_coords(gCurrentCameraID, x, icon->worldPos.y + icon->offsetY, z, &screenX, &screenY, &screenZ);
+                    elementId = icon->surprise.activeElementID;
+                    set_hud_element_render_pos(elementId, screenX + 2, screenY - 15);
+                    draw_hud_element_2(elementId);
+                }
+            } else
+                goto SURPRISE;
+        }
+SURPRISEBACK:
+        if (icon->peril.active) {
+            if (!(icon->flags & STATUS_ICON_FLAG_PERIL)) {
+                if (icon->flags & STATUS_ICON_FLAG_BATTLE || gGameStatusPtr->isBattle != 1) {
+                    clear_hud_element_flags(icon->peril.activeElementID, 2);
+
+                    x = icon->worldPos.x;
+                    z = icon->worldPos.z;
+
+                    get_screen_coords(gCurrentCameraID, x, icon->worldPos.y + icon->offsetY, z, &screenX, &screenY, &screenZ);
+                    elementId = icon->peril.activeElementID;
+                    set_hud_element_render_pos(elementId, screenX + 2, screenY - 16);
+                    draw_hud_element_2(elementId);
+                }
+            } else
+                goto PERIL;
+        }
+PERILBACK:
+        if (icon->danger.active) {
+            if (!(icon->flags & STATUS_ICON_FLAG_DANGER)) {
+                if (icon->flags & STATUS_ICON_FLAG_BATTLE || gGameStatusPtr->isBattle != 1) {
+                    clear_hud_element_flags(icon->danger.activeElementID, 2);
+
+                    x = icon->worldPos.x;
+                    z = icon->worldPos.z;
+
+                    get_screen_coords(gCurrentCameraID, x, icon->worldPos.y + icon->offsetY, z, &screenX, &screenY, &screenZ);
+                    elementId = icon->danger.activeElementID;
+                    set_hud_element_render_pos(elementId, screenX + 2, screenY - 16);
+                    draw_hud_element_2(elementId);
+                }
+            } else
+                goto DANGER;
+        }
+DANGERBACK: ;
+    }
+}
+
+s32 create_status_icon_set(void) {
+    s32 i;
+    HudStatusIcon* icon;
+
+    for (icon = D_800A0F44, i = 0; i < MAX_ICONS; icon++, i++)
+        if (icon->flags == 0)
+            break;
+
+    ASSERT(i < MAX_ICONS)
```

Comment:
It's not necessary, but we end these in semicolons just so they look more like normal statements